### DISCUSSION TITLE
Bun:test - Is it possible to mock `new Date()` without third party-dependencie?

### DISCUSSION THREAD

**Discussion by charnould** [oven-sh/bun#3493] [Q&A] [answered]
`Bun:test` is freaking fast and good!  

Using snapshot testing, I try to validate a function generating an object.  

However, this function adds a `created_at` `key` where `value` is `new Date()`/`Date.now()`.
Hence, every time I run `bun test`, `new Date()` is different and thus my snapshot is outdated.  
Consequence: a failing test.

Is it possible to mock `new Date()` in the same way (or another) you can do it with [`Jest`](https://jestjs.io/docs/timer-mocks).
In other words: "hardcode a certain value of `new Date()`".  
Thanks

**Comment by Jarred-Sumner** [comment]
soon

https://github.com/oven-sh/bun/pull/3501

**Reply by charnould** [reply, marked as answer]
Just saw: https://bun.sh/docs/test/time for `0.6.13` (yet to be released). THANKS. 👍
